test/e2e/upg_e2e.go

```diff
@@ -604,6 +604,98 @@ var _ = ginkgo.Describe("CLI debug commands", func() {
 
 // TODO: validate both binapi and CLI against each other
 var _ = ginkgo.Describe("UPG Binary API", func() {
+	ginkgo.Context("for node-id", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+
+		callSetNodeID := func(req *upf.UpfSetNodeID) (*upf.UpfSetNodeIDReply, error) {
+			reply := &upf.UpfSetNodeIDReply{}
+			err := f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply)
+			return reply, err
+		}
+
+		callGetNodeID := func() (*upf.UpfGetNodeIDReply, error) {
+			req := &upf.UpfGetNodeID{}
+			reply := &upf.UpfGetNodeIDReply{}
+			err := f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply)
+			return reply, err
+		}
+
+		ginkgo.It("sets and retrieves the node-id", func() {
+			ipv4, _ := ip_types.ParseAddress("192.168.42.1")
+			ipv6, _ := ip_types.ParseAddress("2001:0db8:85a3:0000:0000:8a2e:0370:7334")
+
+			fqdnStr := "upg.example.com"
+			fqdn := util.EncodeFQDN(fqdnStr)
+			fqdnLen := uint8(len(fqdn))
+
+			// pass 1: IPv4
+			setReq := &upf.UpfSetNodeID{
+				Type: uint8(upf.UPF_NODE_TYPE_IPv4),
+				IP:   ipv4,
+			}
+			_, setErr := callSetNodeID(setReq)
+			gomega.Expect(setErr).To(gomega.BeNil(), "upf_set_node_id")
```

Comment:
To be consistent with the most of Go code, you need to just call it `err`. Later you can reuse it still.
If you're suing `:=`, just one of the left-side identifiers needs to be "new". So you can write
```go
			_, err := callSetNodeID(setReq)
```
and then still use 
```go
			out, err := f.VPP.Ctl("show upf node-id")
```

If `out` is previously defined, though, you'll have to do
```go
			out, err := f.VPP.Ctl("show upf node-id")
```
IoW: `:=` needs to be defining "some" new vars. But not all vars need to be new.

Also, in Gomega, for `err` checking you write
```go
			gomega.Expect(setErr).NotTo(gomega.HaveOccurred(), "upf_set_node_id")
```

But with the change suggested above, you can actually just use 
```suggestion
			gomega.Expect(callSetNodeID(setReq)).To(gomega.Succeed(), "upf_set_node_id")
```